src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -3741,13 +3741,28 @@ def _ensure_pod_identity_kubenet_consent(network_profile, pod_identity_profile,
     pod_identity_profile.allow_network_plugin_kubenet = True
 
 
+def _fill_defaults_for_pod_identity_exceptions(pod_identity_exceptions):
```

Comment:
better to add some unit test.
@fumingzhang please share some samples.

Reply:
The PR looks good and it seems that the reason for the error could not be covered by the unit tests on the cli side.

But I still agree with liming, it might be difficult to write unit tests for entry functions like `aks_pod_identity_exception_add` or `aks_pod_identity_exception_update` since you may need to mock the command context, client and some requests. However it should be easy to test functions like `_update_addon_pod_identity` or `_fill_defaults_for_pod_identity_exceptions`, and these internal functions are used by many other functions. It is recommended to write unit test cases for pod identity related functions in a separate file.

Reply:
I would suggest we address these issues in follow up PRs. Current change is not related to these proposed tests.